code/modules/mob/living/basic/space_fauna/morph.dm

```diff
@@ -0,0 +1,215 @@
+		. += span_warning("It doesn't look quite right...")
+
+/mob/living/basic/morph/med_hud_set_health()
+	if(isliving(form_typepath))
+		return ..()
+
+	//we hide medical hud while in regular state or an item
+	var/image/holder = hud_list[HEALTH_HUD]
+	holder.icon_state = null
+
+/mob/living/basic/morph/med_hud_set_status()
+	if(isliving(form_typepath))
+		return ..()
+
+	//we hide medical hud while in regular state or an item
+	var/image/holder = hud_list[STATUS_HUD]
+	holder.icon_state = null
+
+/mob/living/basic/morph/death(gibbed)
+	if(!HAS_TRAIT(src, TRAIT_DISGUISED))
+		return ..()
+
+	// the action should handle the rest
+	visible_message(
+		span_warning("[src] twists and dissolves into a pile of green flesh!"),
+		span_userdanger("Your skin ruptures! Your flesh breaks apart! No disguise can ward off de--"),
+	)
+
+	return ..()
+
+/mob/living/basic/morph/can_track(mob/living/user)
+	if(!HAS_TRAIT(src, TRAIT_DISGUISED))
+		return FALSE
+	return ..()
+
+/// Do some more logic for the morph when we disguise through the action.
+/mob/living/basic/morph/proc/on_disguise(mob/living/basic/user, atom/movable/target)
+	SIGNAL_HANDLER
+	// We are now weaker
+	melee_damage_lower = melee_damage_disguised
+	melee_damage_upper = melee_damage_disguised
+	add_movespeed_modifier(/datum/movespeed_modifier/morph_disguised)
+
+	med_hud_set_health()
+	med_hud_set_status() //we're an object honest
+
+	visible_message(
+		span_warning("[src] suddenly twists and changes shape, becoming a copy of [target]!"),
+		span_notice("You twist your body and assume the form of [target]."),
+	)
+
+	form_weakref = WEAKREF(target)
+	form_typepath = target.type
+
+/// Do some more logic for the morph when we undisguise through the action.
+/mob/living/basic/morph/proc/on_undisguise()
+	SIGNAL_HANDLER
+	visible_message(
+		span_warning("[src] suddenly collapses in on itself, dissolving into a pile of green flesh!"),
+		span_notice("You reform to your normal body."),
+	)
+
+	//Baseline stats
+	melee_damage_lower = initial(melee_damage_lower)
+	melee_damage_upper = initial(melee_damage_upper)
+	remove_movespeed_modifier(/datum/movespeed_modifier/morph_disguised)
+
+	med_hud_set_health()
+	med_hud_set_status() //we are no longer an object
+
+	form_weakref = null
+	form_typepath = null
+
+/// Alias for the disguise ability to be used as a keybind.
+/mob/living/basic/morph/proc/trigger_ability(mob/living/basic/source, atom/target)
+	SIGNAL_HANDLER
+
+	// linters hate this if it's not async for some reason even though nothing blocks
+	INVOKE_ASYNC(disguise_ability, TYPE_PROC_REF(/datum/action/cooldown, InterceptClickOn), caller = source, target = target)
+	return COMSIG_MOB_CANCEL_CLICKON
+
+/// Handles the logic for attacking anything.
+/mob/living/basic/morph/proc/pre_attack(mob/living/basic/source, atom/target)
+	SIGNAL_HANDLER
+
+	if(HAS_TRAIT(src, TRAIT_DISGUISED) && (melee_damage_disguised >= 0))
```

Comment:
Am I missing something or why do we have a `melee_damage_disguised` which you can never use?

Reply:
hold over from the simple animal framework, i believe it was meant to be adjustable for special morphs after spawning or for admin schenanigans. i think it's okay for this to stay in case some weird fucker wants to make tables kill people in one hit or something weird in an event

Reply:
maybe im missing something but doesn't this make it impossible for them to vv a morph to make tables kill people in one hit?
The logic here allows you to set a "disguised melee damage" on the morph, but then if you ever try to attack anyone with a damage greater than 0 it cancels the attack making the variable pointless except for explicitly _preventing_ admins from doing that event.

Maybe preventing admins from fucking with it _was_ the point?

Reply:
no that would be because i fucked up the check when i turned it into an early return. it doesn't make sense to prevent admins from fucking with stuff anyhow because it's a sandbox game where wacky stuff should be allowed to happen